examples/neural_networks/plot_random_neural_network.py

```diff
@@ -0,0 +1,155 @@
+"""
+========================================================================
+Impact of varying hyperparameters alpha and weight_scale on toy datasets
+========================================================================
+
+This illustrates how the random neural network behaves when varying the 
+regularization terms, weight_scale and alpha. Regularization usually
+affects the regularity or smoothness of the decision function.
+
+This example generates two plots, corresponding to each of the hyperparameters.
+They control regularization in that they constrain the coefficients of the
+decision function.
+
+If there is high bias - as can be indicated by a high training
+error, decreasing alpha or increasing weight_scale would decrease bias and 
+therefore reduce underfitting. Similarly, if there is high variance - as can be 
+indicated by a low training error, increasing alpha or decreasing weight_scale 
+would decrease variance and therefore reduce overfitting. Therefore, it is best
+to find a balance between bias and variance (or parameter values that are not 
+too high nor too low) by testing a range of values as seen in this example.
+
+"""
+print(__doc__)
+
+
+# Author: Issam H. Laradji
+# License: BSD 3 clause
+
+import numpy as np
+
+from matplotlib import pyplot as plt
+from matplotlib.colors import ListedColormap
+
+from sklearn.cross_validation import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.datasets import make_moons, make_circles, make_classification
+from sklearn.neural_network import RandomBasisFunction
+from sklearn.linear_model import Ridge
+from sklearn.pipeline import make_pipeline
+from sklearn.utils.fixes import expit as logistic_sigmoid
+
+
+# To be removed (no predict_proba in Ridge)
+def predict_proba(clf, x):
+    return logistic_sigmoid(clf.predict(x))
+
+h = .02  # step size in the mesh
+rng = np.random.RandomState(1)
+
+alpha_list = np.logspace(-4, 4, 5)
+weight_scale_list = np.logspace(-2, 2, 5)
+
+
+def plot(names, classifiers, title):
+    X, y = make_classification(n_features=2, n_redundant=0, n_informative=2,
+                               random_state=rng,
+                               n_clusters_per_class=1)
+
+    X += 2 * rng.uniform(size=X.shape)
```

Comment:
Why is this needed? It would look much cleaner without it. If not, could you explain in a comment why?